### PR TITLE
Add JOSS paper

### PR DESCRIPTION
**Pull request by hajdik**
## Purpose
Bring the paper branch into main as the [review](https://github.com/openjournals/joss-reviews/issues/5319) is nearly complete. Also bump the version and add a metadata file for the zenodo build to pick up on.

## Expected time until merged
A day or two so we can do a final proofread

## Type of change
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [x] Other (final paper!)

## Testing
PDF can be viewed [here](https://github.com/mdolab/pygeo/actions/workflows/draft-pdf.yml)

## Checklist
- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: .github/workflows/draft-pdf.yml
Comment (sseraj) [inline .github/workflows/draft-pdf.yml:1]: This file should be removed before merging

File: paper/paper.md
Comment (sseraj) [inline paper/paper.md:74]: The FFD acronym is used before it is defined

File: paper/paper.md
Comment (sseraj) [inline paper/paper.md:125]: The caption should say OpenVSP instead of VSP for consistency with the text

File: paper/paper.md
Comment (sseraj) [inline paper/paper.md:143]: I would change this to:
"... arbitrary triangulated surfaces [@Brelje2020a], illustrated in \autoref{fig:trisurf}."

File: paper/paper.md
Comment (sseraj) [inline paper/paper.md:175]: Is "that contains a geometry parameterization" correct or missing something at the end?

**Review by sseraj — COMMENTED** [review]
I left some minor comments

File: .zenodo.json
Comment (nwu63) [inline .zenodo.json:2]: Also needs title here

File: paper/paper.md
Comment (nwu63) [inline paper/paper.md:168]: The `(Sec.~6.7)` is not rendered correctly.

**Review by sseraj — APPROVED** [review]

File: paper/paper.md
Comment (nwu63) [inline paper/paper.md:149]: MPI is not defined anywhere

Comment (nwu63) [reply paper/paper.md:149]: Same for CFD.

**Review by nwu63 — COMMENTED** [review]
LGTM other than two minor things. Once fixed (and required approvals are obtained) we can remove the GHA file, and merge.

File: paper/paper.md
Comment (sseraj) [inline paper/paper.md:168]: One other point that I missed earlier: I don't think "implicit analytic methods" is correct. This section of the book refers to the direct and adjoint methods, but the FFD parameterization does not solve any linear systems. The FFD parameterization is an explicit function that is differentiated mostly by hand (and some complex step).

**Review by nwu63 — APPROVED** [review]
LGTM, don't forget to make a release after merged!

**Pull request merged**
